solidity/test/rewards/TestRewardUpgrades.js

```diff
@@ -168,6 +168,60 @@ describe("Rewards/Upgrades", () => {
             )
         })
 
+        it("allocates interval before finalize rewards upgrade", async () => {
+            await time.increase(termLength + 1) // interval 0 ends
+
+            await rewards.initiateRewardsUpgrade(
+                newRewards.address,
+                {from: owner}
+            )
+
+            await time.increase(termLength + 1) // interval 1 ends
+            await rewards.setCloseTime(timestamps[2])
+
+            await rewards.allocateRewards(1)
+            await rewards.finalizeRewardsUpgrade({from: owner})
+
+            const upgradeInitiatedTimestamp = await rewards.upgradeInitiatedTimestamp()
+            const upgradeFinalizedTimestamp = await rewards.upgradeFinalizedTimestamp()
+
+            expect(upgradeInitiatedTimestamp).to.eq.BN(0)
+            expect(upgradeFinalizedTimestamp).not.to.eq.BN(0)
```

Comment:
I'd extract those assertions into a separate test ensuring timestamps are correctly updated and instead, in this test, I would check allocations for intervals, just like in the previous one. We are testing interval allocations so let's assert interval allocations.

Reply:
This test was rather about passing finalization process (hence checking the timestamp, cause they are at the end of the finalization function) and not allocations. I thought we had plenty of tests just for allocations values and that's why I didn't add allocations here. Anyway, changed now.